packages/ext-taquito/src/readProviderWrapper.ts

```diff
@@ -0,0 +1,187 @@
+import {
+  BlockResponse,
+  EntrypointsResponse,
+  MichelsonV1Expression,
+  SaplingDiffResponse, ScriptedContracts,
+} from '@taquito/rpc';
+import {
+  BigMapQuery,
+  BlockIdentifier,
+  SaplingStateQuery,
+  TzReadProvider
+} from '@taquito/taquito';
+import {BigNumber} from 'bignumber.js';
+
+import {TzktExtension} from './index';
+
+
+export class ReadProviderWrapper implements TzReadProvider {
+  constructor(private readProvider: TzReadProvider, private tzktExtension: TzktExtension) {}
+
+  async getEntrypoints(contract: string): Promise<EntrypointsResponse> {
+    const apiResponse = this.tzktExtension.getEntrypoints(contract);
+
+    if (apiResponse) {
+      return apiResponse;
+    }
+
+    return this.readProvider.getEntrypoints(contract);
```

Comment:
No need to fallback in all methods, just those where we do not have a replacement in TzKT